apis/projectcontour/v1/httpproxy.go

```diff
@@ -725,6 +725,36 @@ type HeaderHashOptions struct {
 	HeaderName string `json:"headerName,omitempty"`
 }
 
+// CookieHashOptions contains options to configure a HTTP Cookie on which to hash
+// used in request attribute hash based load balancing.
+type CookieHashOptions struct {
+	// UseDefaultSessionAffinity
+	UseDefaultSessionAffinity bool `json:"useDefaultSessionAffinity,omitempty"`
+
+	// CookieName is the name of the HTTP Cookie that will be used to
+	// calculate the hash key. If the cookie specified is not present on a
+	// request and the Passive field is true, no hash will be produced.
+	// If the cookie specified is not present on a request and the Passive
+	// field is false, a hash will be produced and sent in the Set-Cookie
+	// header of the response.
```

Comment:
Maybe a short statement here about what happens when the cookie is set? Something like "If the cookie specified is present, then that value will be the value used to generate a hash to a backend; as long as no backends change, the traffic will flow consistently to a single backend."